### ISSUE TITLE
Infer UseSSL value for SPWebApplication from URL

### ISSUE THREAD

**Issue by BrianFarnhill** [PowerShell/SharePointDsc#424]
**Details of the scenario you try and problem that is occurring:**

Currently [SPWebApplication](https://github.com/PowerShell/SharePointDsc/wiki/SPWebApplication) requires you to set the "UseSSL" property manually, and if you set this to true but provide a URL property that does not start with "https" SharePoint will throw an error. We should be able to detect the type of URL that is provided and automatically set the UseSSL attribute based on that, taking the need to do this away from the author and removing the chance of it being mistakenly entered wrong.

**The DSC configuration that is using the resource:**

```
SPWebApplication SSLFailingWebApp
{
    Url = "http://sharepoint.contoso.com"
    UseSSL = $true
}
```

**Version of the Operating System and PowerShell the DSC Target Node is running:**

n/a

**Version of the DSC module you're using:**

SharePointDsc 1.3

**Comment by ykuijs** [comment]
I am on it!

**Comment by MikeLacher448** [comment]
I had a use case today where my customer wanted to offload SSL to an F5. The way the F5 proxying works, I had to name the web application URL with https, even though it use SSL was set to false, and used port 80. Otherwise, the 302 redirects coming from SharePoint used http in the redirect locations.

**Comment by BrianFarnhill** [comment]
@MikeLacher448 I think we need to check this, because I'm actually pretty sure if you try to create a web app with a https address and you don't use the UseSSL flag it gives an error - we really only rely on this when creating the web apps. Happy to explore the combinations of what New-SPWebApplication will take and make sure we don't exclude any scenarios though. Thanks for raising this!

**Comment by MikeLacher448** [comment]
confirmed, you do get an error trying to create or extend a web application with https and not specifying usessl. In my particular case, the web app had to be created with http://, with the public URLs updated afterwards. This particular case will cause the consistency checks to fail as they check for a http:// web application and don't find it.

**Comment by BrianFarnhill** [comment]
@MikeLacher448 Ahhh, ok so at this point we know things go south in terms of DSC when you start modifying the default zone URLs for the web app, so we don't recommended doing that if you want to manage things with DSC.

**Comment by ykuijs** [comment]
Implemented change for a 2.0 PR

**Comment by ykuijs** [comment]
Fix merged into 2.0 branche
